daliuge-engine/dlg/drop.py

```diff
@@ -1931,11 +1931,18 @@ def async_execute(self):
         # Return immediately, but schedule the execution of this app
         # If we have been given a thread pool use that
         if hasattr(self, "_tp"):
-            self._tp.apply_async(self.execute)
+            self._tp.apply_async(self._execute_and_log_exception)
         else:
-            t = threading.Thread(target=self.execute)
+            t = threading.Thread(target=self._execute_and_log_exception)
             t.daemon = 1
             t.start()
+            return t
```

Comment:
This is a nice touch, together with then joining the thread. I wrote this a looooooong time ago, and probably what should be done is to remove this path, use a Drop._tp thread pool instead of a per-instance one, and set it up at the start of the tests.

Reply:
If that's the case, can we assume that `_tp` should exist? It looks like this is provided quite dynamically and if we want to be sure it exists it probably should be passed in at construction. But yep won't make that change here since this is small bugfix ticket :slightly_smiling_face: